api/src/test/java/org/openmrs/test/jupiter/BaseContextSensitiveTest.java

```diff
@@ -346,10 +347,17 @@ public Properties getRuntimeProperties() {
 		runtimeProperties.setProperty(ModuleConstants.IGNORE_CORE_MODULES_PROPERTY, "true");
 		
 		try {
-			File tempappdir = File.createTempFile("appdir-for-unit-tests-", "");
+			final File tempappdir = File.createTempFile("appdir-for-unit-tests-", "");
 			tempappdir.delete(); // so we can make it into a directory
 			tempappdir.mkdir(); // turn it into a directory
-			tempappdir.deleteOnExit(); // clean up when we're done with tests
+			// clean up when we're done with tests
+			Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+				try {
+					FileUtils.deleteDirectory(tempappdir);
+				}
+				catch (IOException ignored) {
+				}
+			}));
```

Comment:
So, here, based on @teleivo's comments, we'd want to replace this with a static class variable. After the line that looks like this: `protected static Properties runtimeProperties;`, we'd want to add

```junit
@TempDir
private static File tempappdir;
```

Reply:
@ibacher, @teleivo I did as suggested but run into this error https://pastebin.com/uAtB1VhR 
	```
/**
	 * cached runtime properties
	 */
	protected static Properties runtimeProperties;
	
	@TempDir
	private static File tempappdir;
```

Reply:
@jwnasambu the error you get says the field must not be private. so try to make it protected and if that does not work make it public